Include/fileutils.h

```diff
@@ -81,6 +81,17 @@ PyAPI_FUNC(int) _Py_EncodeLocaleEx(
 #ifndef Py_LIMITED_API
 PyAPI_FUNC(PyObject *) _Py_device_encoding(int);
 
+#if defined(MS_WINDOWS) || defined(__APPLE__)
+/* On Windows, the count parameter of read() is an int.
+   macOS fails when reading or writing more than 2GB, see #24658
+*/
+#define _PY_READ_MAX  INT_MAX
+#define _PY_WRITE_MAX INT_MAX
+#else
+#define _PY_READ_MAX  PY_SSIZE_T_MAX
+#define _PY_WRITE_MAX PY_SSIZE_T_MAX
```

Comment:
I suggest to move the comment here:

        /* write() should truncate count to PY_SSIZE_T_MAX, but it's safer
         * to do it ourself to have a portable behaviour. */

Reply:
like that?
```
#define _PY_READ_MAX  INT_MAX
#define _PY_WRITE_MAX INT_MAX
#else
#define _PY_READ_MAX  PY_SSIZE_T_MAX
#define _PY_WRITE_MAX PY_SSIZE_T_MAX
/* write() should truncate count to PY_SSIZE_T_MAX, but it's safer
 * to do it ourself to have a portable behaviour */ 
#endif
```

because I am not sure to understand your suggestion.